internal/obsreportconfig/obsreportconfig.go

```diff
@@ -15,6 +15,8 @@
 package obsreportconfig // import "go.opentelemetry.io/collector/internal/obsreportconfig"
 
 import (
```

Comment:
This change is the only behavioural change being made within the PR to ensure that access is guarded by atomic access. Currently it will eventually consistent over the lifetime of the application.

Reply:
Can this be extracted separately?

Reply:
Yeah, I noticed that this impacts a lot changes.

I was on the fence if I should be its own PR

Reply:
Have made the PR here @bogdandrutu https://github.com/open-telemetry/opentelemetry-collector/pull/4549